pdg/src/builder.rs

```diff
@@ -1,136 +1,148 @@
 use crate::graph::{Func, Graph, GraphId, Graphs, Node, NodeId, NodeKind};
 use bincode;
-use c2rust_analysis_rt::events::{Event, EventKind};
+use c2rust_analysis_rt::events::{Event, EventKind, Pointer};
 use c2rust_analysis_rt::mir_loc::{EventMetadata, Metadata, TransferKind};
-use c2rust_analysis_rt::{mir_loc, MirLoc, MirPlace, MirProjection};
+use c2rust_analysis_rt::{mir_loc, MirLoc};
 use log;
 use rustc_data_structures::fingerprint::Fingerprint;
 use rustc_hir::def_id::DefPathHash;
-use rustc_middle::mir::{Field, Local};
-use std::borrow::Borrow;
 use std::collections::HashMap;
-use std::fs::File;
-use std::io::BufReader;
+use std::fs::{self, File};
+use std::io::{self, BufReader};
+use std::iter;
+use std::path::Path;
 
-pub fn read_event_log(path: String) -> Vec<Event> {
-    let file = File::open(path).unwrap();
-    let mut events = vec![];
+pub fn read_event_log(path: &Path) -> io::Result<Vec<Event>> {
+    let file = File::open(path)?;
     let mut reader = BufReader::new(file);
-    loop {
-        match bincode::deserialize_from(&mut reader) {
-            Ok(e) => events.push(e),
-            _ => break,
-        }
-    }
-    events
+    let events = iter::from_fn(|| bincode::deserialize_from(&mut reader).ok()).collect::<Vec<_>>();
```

Comment:
nice, wasn't aware of `from_fn`

Reply:
Yeah, it's nice.  There are so many cool iterator functions and impls.